Sources/MapboxNavigation/ImageRepository.swift

```diff
@@ -39,8 +48,11 @@ class ImageRepository {
             return
         }
 
-        let _ = imageDownloader.downloadImage(with: imageURL, completion: { [weak self] (image, data, error) in
-            guard let strongSelf = self, let image = image else {
+        let request = urlRequest(with: imageURL)
+        let _ = requestDownloader.download(with: request, completion: { [weak self] (cachedResponse, error) in
+            guard let strongSelf = self,
+                  let data = cachedResponse?.data,
+                  let image = UIImage(data: data, scale: UIScreen.main.scale) else {
```

Comment:
We may make it another ticket to handle all the `UIScreen.main.scale` for the new beta version.

Reply:
So from `Directions` side to our side, we all request images based on the `UIScreen.main.scale` for iOS. And in our side, we also parse the data to image with the `UIScreen.main.scale`. so it should be fine because they're consistent.

Reply:
Scale factor might be different on iOS and CarPlay. `ImageRepository` should be aware of that to correctly generate image. In case of CarPlay we can use `UIScreen.mainCarPlay` extension.

Reply:
Because different platforms may have different scale factors. The `SpriteRepository` is used as a singleton across the platform. #4001 will add the ability to cache and provide the images for different platform with its scale factor. The issue will be solved by #4001